### PR TITLE
Bug 2076831: fixing client readyz leak

### PR DESCRIPTION
**Pull request by tjungblu**

### CODE REVIEW

File: pkg/cmd/readyz/readyz.go [hunk 1]
```diff
@@ -147,6 +147,10 @@ func (r *readyzOpts) Run() error {
 func (r *readyzOpts) getReadyzHandlerFunc(ctx context.Context) http.HandlerFunc {
 	return func(w http.ResponseWriter, _ *http.Request) {
 		etcdClient, err := r.newETCD3Client(ctx, r.targetEndpoint)
+		defer func() {
+			err = etcdClient.Close()
+			klog.V(2).Infof("error closing etcd client: %v", err)
```
Comment (mfojtik) [inline pkg/cmd/readyz/readyz.go:152]: `if err != ....` ?

**Comment by mfojtik** [comment]
@hasbro17 this is fixing memory leak

File: pkg/cmd/readyz/readyz.go [hunk 2]
```diff
@@ -147,6 +147,12 @@ func (r *readyzOpts) Run() error {
 func (r *readyzOpts) getReadyzHandlerFunc(ctx context.Context) http.HandlerFunc {
 	return func(w http.ResponseWriter, _ *http.Request) {
 		etcdClient, err := r.newETCD3Client(ctx, r.targetEndpoint)
+		defer func() {
+			err = etcdClient.Close()
```
Comment (deads2k) [inline pkg/cmd/readyz/readyz.go:151]: if etcdClient is nil, will this panic?

File: pkg/cmd/readyz/readyz.go [hunk 3]
```diff
@@ -147,6 +147,13 @@ func (r *readyzOpts) Run() error {
 func (r *readyzOpts) getReadyzHandlerFunc(ctx context.Context) http.HandlerFunc {
 	return func(w http.ResponseWriter, _ *http.Request) {
 		etcdClient, err := r.newETCD3Client(ctx, r.targetEndpoint)
+		defer func() {
+			if etcdClient != nil {
```
Comment (deads2k) [inline pkg/cmd/readyz/readyz.go:151]: normally we 

```
if etcdClient == nil{
  return 
}
if err := etcdClient.Close()
```

**Comment by deads2k** [comment]
infra 

/retest-required

**Pull request merged**